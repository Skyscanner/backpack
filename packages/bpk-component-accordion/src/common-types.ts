
export const ACCORDION_TYPES = {
  default: 'default',
  surfaceDefault: 'surface-default',
  surfaceLowContrast: 'surface-low-contrast',
} as const;

export type AccordionType = (typeof ACCORDION_TYPES)[keyof typeof ACCORDION_TYPES];
