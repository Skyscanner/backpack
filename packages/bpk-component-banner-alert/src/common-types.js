/*
 * Backpack - Skyscanner's Design System
 *
 * Copyright 2016 Skyscanner Ltd
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *   http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
/* @flow strict */
import PropTypes from 'prop-types';
import { type Node, type StatelessFunctionalComponent } from 'react';

export const ALERT_TYPES = {
  PRIMARY: 'primary',
  SUCCESS: 'success',
  WARN: 'warn',
  ERROR: 'error',
  NEUTRAL: 'neutral',
  EVENT: 'event', // DEPRECATED
};

export type AlertTypeValue = $Values<typeof ALERT_TYPES>;

export type CommonProps = {
  type: AlertTypeValue,
  message: Node,
  animateOnEnter: boolean,
  animateOnLeave: boolean,
  show: boolean,
  bannerClassName: ?string,
  icon: ?StatelessFunctionalComponent<any>,
};

export type OnExpandToggleHandler = ?(boolean) => void;
export type OnDismissHandler = ?() => void;
export type OnHideHandler = ?() => void;

export const COMMON_PROP_TYPES = {
  type: PropTypes.oneOf([
    ALERT_TYPES.PRIMARY,
    ALERT_TYPES.SUCCESS,
    ALERT_TYPES.WARN,
    ALERT_TYPES.ERROR,
    ALERT_TYPES.NEUTRAL,
    ALERT_TYPES.EVENT,
  ]).isRequired,
  message: PropTypes.node.isRequired,
  animateOnEnter: PropTypes.bool,
  animateOnLeave: PropTypes.bool,
  show: PropTypes.bool,
  bannerClassName: PropTypes.string,
  icon: PropTypes.func,
};

export const COMMON_DEFAULT_PROPS = {
  animateOnEnter: false,
  animateOnLeave: false,
  show: true,
  bannerClassName: null,
  icon: null,
};
