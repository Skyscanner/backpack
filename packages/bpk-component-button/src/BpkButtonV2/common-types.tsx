/*
 * Backpack - Skyscanner's Design System
 *
 * Copyright 2016 Skyscanner Ltd
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *   http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

import type { MouseEvent, ReactNode } from 'react';

export const BUTTON_TYPES = {
  primary: 'primary',
  primaryOnDark: 'primary-on-dark',
  primaryOnLight: 'primary-on-light',
  secondary: 'secondary',
  secondaryOnDark: 'secondary-on-dark',
  destructive: 'destructive',
  featured: 'featured',
  link: 'link',
  linkOnDark: 'link-on-dark',
} as const;

export const SIZE_TYPES = {
  small: 'small',
  large: 'large',
} as const;

export type ButtonType = (typeof BUTTON_TYPES)[keyof typeof BUTTON_TYPES];
export type SizeType = (typeof SIZE_TYPES)[keyof typeof SIZE_TYPES];

export type Props = {
  children: string | ReactNode;
  type?: ButtonType;
  size?: SizeType;
  className?: string | null;
  disabled?: boolean;
  iconOnly?: boolean;
  onClick?: (event: MouseEvent<HTMLButtonElement & HTMLAnchorElement>) => void;
  rel?: string | undefined;
  submit?: boolean;
  href?: string | null;
  blank?: boolean;
  [rest: string]: any; // Inexact rest. See decisions/inexact-rest.md
};
