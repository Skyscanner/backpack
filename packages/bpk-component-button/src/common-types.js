/*
 * Backpack - Skyscanner's Design System
 *
 * Copyright 2016 Skyscanner Ltd
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *   http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

/* @flow strict */

import { type Node } from 'react';
import PropTypes from 'prop-types';

export type Props = {
  children: Node,
  href: ?string,
  className: ?string,
  disabled: boolean,
  onClick: ?(event: SyntheticEvent<>) => mixed,
  submit: boolean,
  large: boolean,
  iconOnly: boolean,
  blank: boolean,
  rel: ?string,
};

const propTypes = {
  children: PropTypes.node.isRequired,
  href: PropTypes.string,
  className: PropTypes.string,
  disabled: PropTypes.bool,
  onClick: PropTypes.func,
  submit: PropTypes.bool,
  large: PropTypes.bool,
  iconOnly: PropTypes.bool,
  blank: PropTypes.bool,
  rel: PropTypes.string,
};

const defaultProps = {
  href: null,
  className: null,
  disabled: false,
  onClick: null,
  submit: false,
  large: false,
  iconOnly: false,
  blank: false,
  rel: null,
};

export { propTypes, defaultProps };
