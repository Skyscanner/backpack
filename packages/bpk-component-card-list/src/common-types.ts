/*
 * Backpack - Skyscanner's Design System
 *
 * Copyright 2016 Skyscanner Ltd
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *   http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

import type { Dispatch, ReactElement, SetStateAction } from 'react';

const LAYOUTS = {
  grid: 'grid',
  stack: 'stack',
} as const;

type DesktopLayouts = typeof LAYOUTS.grid;
type MobileLayouts = typeof LAYOUTS.stack;

const ACCESSORY_TYPES = {
  Expand: 'expand',
  Button: 'button',
} as const;

type ExpandProps = {
  children: string | ReactElement;
  collapsed: boolean;
  hideContent: () => void;
  setCollapsed: Dispatch<SetStateAction<boolean>>;
  showContent: () => void;
};

type CardListBaseProps = {
  title: string;
  description?: string;
  buttonText?: string;
  onButtonClick?: () => void;
  buttonHref?: string;
  layoutMobile: MobileLayouts;
  layoutDesktop: DesktopLayouts;
  initiallyShownCards?: number;
  cardList: ReactElement[];
  expandText?: string;
  accessory?: (typeof ACCESSORY_TYPES)[keyof typeof ACCESSORY_TYPES];
};

type CardListGridStackProps = {
  children: ReactElement[];
  accessory?: typeof ACCESSORY_TYPES.Expand | typeof ACCESSORY_TYPES.Button;
  expandText?: string;
  buttonText?: string;
  onButtonClick?: () => void;
  initiallyShownCards: number;
  layout: typeof LAYOUTS.grid | typeof LAYOUTS.stack;
};

type CardListProps = CardListBaseProps;

export default CardListProps;
export { LAYOUTS, ACCESSORY_TYPES };
export type {
  DesktopLayouts,
  MobileLayouts,
  CardListGridStackProps,
  ExpandProps,
};
