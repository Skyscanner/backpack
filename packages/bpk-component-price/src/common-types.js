/*
 * Backpack - Skyscanner's Design System
 *
 * Copyright 2016 Skyscanner Ltd
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *   http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

/* @flow strict */

import PropTypes from 'prop-types';

export const LAYOUTS = {
  small: 'small',
  large: 'large',
};

export type Props = {
  title: string,
  layout: string,
  className: ?string,
  subtitle: ?string,
  description: ?string,
};

export const propTypes = {
  title: PropTypes.string.isRequired,
  layout: PropTypes.oneOf(Object.keys(LAYOUTS)),
  className: PropTypes.string,
  subtitle: PropTypes.string,
  description: PropTypes.string,
};

export const defaultProps = {
  className: null,
  subtitle: null,
  description: null,
};
