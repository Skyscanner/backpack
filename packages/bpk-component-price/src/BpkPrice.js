/*
 * Backpack - Skyscanner's Design System
 *
 * Copyright 2016 Skyscanner Ltd
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *   http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
/* @flow strict */

import PropTypes from 'prop-types';
import React from 'react';

import { cssModules } from '../../bpk-react-utils';
import BpkText, { TEXT_STYLES } from '../../bpk-component-text';

import STYLES from './BpkPrice.module.scss';
import { SIZES, ALIGNS } from './common-types';

type Props = {
  title: string,
  size: $Values<typeof SIZES>,
  align: $Values<typeof ALIGNS>,
  className: ?string,
  subtitle: ?string,
  description: ?string,
};

const getClassName = cssModules(STYLES);

const BpkPrice = (props: Props) => {
  const { align, className, description, size, subtitle, title, ...rest } =
    props;

  const isSmall = size === SIZES.small;
  const isAlignRight = align === ALIGNS.right;

  return (
    <div
      className={getClassName(
        'bpk-price',
        isAlignRight && 'bpk-price--right',
        className,
      )}
      // $FlowFixMe[cannot-spread-inexact] - inexact rest. See 'decisions/flowfixme.md'.
      {...rest}
    >
      {subtitle && (
        <BpkText
          className={getClassName('bpk-price__subtitle')}
          textStyle={isSmall ? TEXT_STYLES.xs : TEXT_STYLES.sm}
          tagName="span"
        >
          {subtitle}
        </BpkText>
      )}
      <div
        className={isAlignRight && getClassName('bpk-price__column-container')}
      >
        <BpkText
          textStyle={isSmall ? TEXT_STYLES.heading4 : TEXT_STYLES.xxl}
          className={getClassName(
            'bpk-price__title',
            !isAlignRight && 'bpk-price__spacing',
          )}
          tagName="span"
        >
          {title}
        </BpkText>
        {description && (
          <BpkText
            textStyle={isSmall ? TEXT_STYLES.xs : TEXT_STYLES.sm}
            tagName="span"
            className={getClassName('bpk-price__description')}
          >
            {description}
          </BpkText>
        )}
      </div>
    </div>
  );
};

BpkPrice.propTypes = {
  title: PropTypes.string.isRequired,
  size: PropTypes.oneOf(Object.keys(SIZES)),
  align: PropTypes.oneOf(Object.keys(ALIGNS)),
  className: PropTypes.string,
  subtitle: PropTypes.string,
  description: PropTypes.string,
};

BpkPrice.defaultProps = {
  size: SIZES.small,
  align: ALIGNS.left,
  className: null,
  subtitle: null,
  description: null,
};

export default BpkPrice;
