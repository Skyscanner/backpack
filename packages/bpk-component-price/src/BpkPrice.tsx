/*
 * Backpack - Skyscanner's Design System
 *
 * Copyright 2016 Skyscanner Ltd
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *   http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
/* @flow strict */

import PropTypes from 'prop-types';
// @ts-expect-error TS(2305): Module '"react"' has no exported member 'Node'.
import type { Node } from 'react';

import BpkText, { TEXT_STYLES } from '../../bpk-component-text';
import { cssModules } from '../../bpk-react-utils';

import { SIZES, ALIGNS } from './common-types';

// @ts-expect-error TS(2307): Cannot find module './BpkPrice.module.scss' or its... Remove this comment to see the full error message
import STYLES from './BpkPrice.module.scss';

type Props = {
  price: string,
  // @ts-expect-error TS(2304): Cannot find name '$Values'.
  size: $Values<typeof SIZES>,
  // @ts-expect-error TS(2304): Cannot find name '$Values'.
  align: $Values<typeof ALIGNS>,
  // @ts-expect-error TS(8020): JSDoc types can only be used inside documentation ... Remove this comment to see the full error message
  className: ?string,
  // @ts-expect-error TS(8020): JSDoc types can only be used inside documentation ... Remove this comment to see the full error message
  leadingText: ?string,
  /**
   * **Experimental** This prop is experimental and subject to change.
   * Use with caution.
   */
  // @ts-expect-error TS(8020): JSDoc types can only be used inside documentation ... Remove this comment to see the full error message
  leadingClassName: ?string,
  // @ts-expect-error TS(8020): JSDoc types can only be used inside documentation ... Remove this comment to see the full error message
  trailingText: ?string,
  // @ts-expect-error TS(8020): JSDoc types can only be used inside documentation ... Remove this comment to see the full error message
  previousPrice: ?string,
  icon?: Node,
  dataAttributes?: Record<string, string>,
};

const getClassName = cssModules(STYLES);

// @ts-expect-error TS(2304): Cannot find name '$Values'.
const getPriceTextStyle = (size: $Values<typeof SIZES>) => {
  if (size === SIZES.small) {
    return TEXT_STYLES.heading4;
  }

  if (size === SIZES.large) {
    return TEXT_STYLES.xxl;
  }

  return TEXT_STYLES.heading5;
};

// @ts-expect-error TS(2304): Cannot find name '$Values'.
const getDefaultTextStyle = (size: $Values<typeof SIZES>) => {
  if (size === SIZES.large) {
    return TEXT_STYLES.sm;
  }

  return TEXT_STYLES.xs;
};

const BpkPrice = (props: Props) => {
  const {
    align,
    className,
    dataAttributes,
    icon,
    leadingClassName,
    leadingText,
    previousPrice,
    price,
    size,
    trailingText,
    ...rest
  } = props;

  const defaultTextStyle = getDefaultTextStyle(size);
  const priceTextStyle = getPriceTextStyle(size);
  const isAlignRight = align === ALIGNS.right;

  const getTrailingTextNode = () => {
    if (!trailingText) {
      return null;
    }
    const textNode = (
      <BpkText textStyle={defaultTextStyle} tagName="span">
        {trailingText}
      </BpkText>
    );
    // When aligned right, use a <div> as the text appears on a separate line.
    // When aligned left, use a <span> as the test is inline.
    return isAlignRight ? (
      <div className={getClassName('bpk-price__trailing')}>{textNode}</div>
    ) : (
      <span className={getClassName('bpk-price__trailing')}>{textNode}</span>
    );
  };

  return (
    <div
      className={getClassName(
        'bpk-price',
        isAlignRight && 'bpk-price--right',
        className,
      )}
      // $FlowFixMe[cannot-spread-inexact] - inexact rest. See 'decisions/flowfixme.md'.
      {...rest}
    >
      <div
        className={getClassName(
          previousPrice && 'bpk-price__leading',
          isAlignRight && 'bpk-price__leading--right',
          leadingClassName,
        )}
      >
        {previousPrice && (
          <span className={getClassName('bpk-price__previous-price')}>
            <BpkText textStyle={defaultTextStyle} tagName="span">
              {previousPrice}
            </BpkText>
          </span>
        )}
        {previousPrice && leadingText && (
          <span className={getClassName('bpk-price__separator')}>
            <BpkText textStyle={defaultTextStyle} tagName="span">
              &#67871;
            </BpkText>
          </span>
        )}
        {leadingText && (
          <BpkText textStyle={defaultTextStyle} tagName="span">
            {leadingText}
          </BpkText>
        )}
      </div>
      <div className={getClassName(isAlignRight && 'bpk-price__main--right')}>
        <span
          className={getClassName(
            'bpk-price__price',
            // When aligning right, we use the gap property to add space between the price and the icon.
            // When aligning left, we use the ::after pseudo-element instead to achieve the desired wrapping behaviour.
            !isAlignRight && 'bpk-price__spacing',
          )}
        >
          <BpkText
            textStyle={priceTextStyle}
            tagName="span"
            {...dataAttributes}
          >
            {price}
          </BpkText>
        </span>
        {icon && (
          <span
            className={getClassName(
              'bpk-price__icon',
              !isAlignRight && 'bpk-price__spacing',
            )}
          >
            {icon}
          </span>
        )}
        {!isAlignRight && getTrailingTextNode()}
      </div>
      {isAlignRight && getTrailingTextNode()}
    </div>
  );
};

BpkPrice.propTypes = {
  price: PropTypes.string.isRequired,
  size: PropTypes.oneOf(Object.keys(SIZES)),
  align: PropTypes.oneOf(Object.keys(ALIGNS)),
  className: PropTypes.string,
  leadingText: PropTypes.string,
  trailingText: PropTypes.string,
  previousPrice: PropTypes.string,
  leadingClassName: PropTypes.string,
  dataAttributes: PropTypes.objectOf(PropTypes.string),
};

BpkPrice.defaultProps = {
  size: SIZES.small,
  align: ALIGNS.left,
  className: null,
  leadingText: null,
  trailingText: null,
  previousPrice: null,
  leadingClassName: null,
  dataAttributes: {},
};

export default BpkPrice;
