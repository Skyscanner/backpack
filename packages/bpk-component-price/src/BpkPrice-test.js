/*
 * Backpack - Skyscanner's Design System
 *
 * Copyright 2016 Skyscanner Ltd
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *   http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
/* @flow strict */

import { render } from '@testing-library/react';

import BpkPrice from './BpkPrice';
import { ALIGNS, SIZES } from './common-types';

const price = '£1,830';
const previousPrice = '£2,000';
const leadingText = 'from';
const trailingText = 'per day';
let props;

describe.each([
  [SIZES.small, ALIGNS.left],
  [SIZES.large, ALIGNS.left],
  [SIZES.small, ALIGNS.right],
])(`%s %s view`, (size, align) => {
  beforeEach(() => {
    props = {
      price,
      size,
      align,
    };
  });

  it('should render correctly', () => {
    const { asFragment } = render(<BpkPrice {...props} />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('should support trailing text attribute', () => {
    const { asFragment } = render(
      <BpkPrice {...props} trailingText={trailingText} />,
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it('should support leading text attribute', () => {
    const { asFragment } = render(
      <BpkPrice {...props} leadingText={leadingText} />,
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it('should support previous price attribute', () => {
    const { asFragment } = render(
      <BpkPrice
        {...props}
        previousPrice={previousPrice}
        leadingText={leadingText}
        trailingText={trailingText}
      />,
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it('should support custom class names', () => {
    const { asFragment } = render(
      <BpkPrice {...props} className="custom-classname" />,
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
