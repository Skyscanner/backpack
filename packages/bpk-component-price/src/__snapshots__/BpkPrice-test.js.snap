// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`large left view should render correctly 1`] = `
<DocumentFragment>
  <div
    class="bpk-price"
  >
    <div
      class=""
    />
    <div>
      <span
        class="bpk-text bpk-text--xxl bpk-price__price bpk-price__spacing"
      >
        £1,830
      </span>
    </div>
  </div>
</DocumentFragment>
`;

exports[`large left view should support custom class names 1`] = `
<DocumentFragment>
  <div
    class="bpk-price custom-classname"
  >
    <div
      class=""
    />
    <div>
      <span
        class="bpk-text bpk-text--xxl bpk-price__price bpk-price__spacing"
      >
        £1,830
      </span>
    </div>
  </div>
</DocumentFragment>
`;

exports[`large left view should support leading text attribute 1`] = `
<DocumentFragment>
  <div
    class="bpk-price"
  >
    <div
      class=""
    >
      <span
        class="bpk-text bpk-text--sm"
      >
        from
      </span>
    </div>
    <div>
      <span
        class="bpk-text bpk-text--xxl bpk-price__price bpk-price__spacing"
      >
        £1,830
      </span>
    </div>
  </div>
</DocumentFragment>
`;

exports[`large left view should support previous price attribute 1`] = `
<DocumentFragment>
  <div
    class="bpk-price"
  >
    <div
      class="bpk-price__leading"
    >
      <span
        class="bpk-text bpk-text--sm bpk-price__previous-price"
      >
        £2,000
      </span>
      <span
        class="bpk-text bpk-text--sm bpk-price__separator"
      >
        𐤟
      </span>
      <span
        class="bpk-text bpk-text--sm"
      >
        from
      </span>
    </div>
    <div>
      <span
        class="bpk-text bpk-text--xxl bpk-price__price bpk-price__spacing"
      >
        £1,830
      </span>
      <span
        class="bpk-text bpk-text--sm bpk-price__trailing"
      >
        per day
      </span>
    </div>
  </div>
</DocumentFragment>
`;

exports[`large left view should support trailing text attribute 1`] = `
<DocumentFragment>
  <div
    class="bpk-price"
  >
    <div
      class=""
    />
    <div>
      <span
        class="bpk-text bpk-text--xxl bpk-price__price bpk-price__spacing"
      >
        £1,830
      </span>
      <span
        class="bpk-text bpk-text--sm bpk-price__trailing"
      >
        per day
      </span>
    </div>
  </div>
</DocumentFragment>
`;

exports[`small left view should render correctly 1`] = `
<DocumentFragment>
  <div
    class="bpk-price"
  >
    <div
      class=""
    />
    <div>
      <span
        class="bpk-text bpk-text--heading-4 bpk-price__price bpk-price__spacing"
      >
        £1,830
      </span>
    </div>
  </div>
</DocumentFragment>
`;

exports[`small left view should support custom class names 1`] = `
<DocumentFragment>
  <div
    class="bpk-price custom-classname"
  >
    <div
      class=""
    />
    <div>
      <span
        class="bpk-text bpk-text--heading-4 bpk-price__price bpk-price__spacing"
      >
        £1,830
      </span>
    </div>
  </div>
</DocumentFragment>
`;

exports[`small left view should support leading text attribute 1`] = `
<DocumentFragment>
  <div
    class="bpk-price"
  >
    <div
      class=""
    >
      <span
        class="bpk-text bpk-text--xs"
      >
        from
      </span>
    </div>
    <div>
      <span
        class="bpk-text bpk-text--heading-4 bpk-price__price bpk-price__spacing"
      >
        £1,830
      </span>
    </div>
  </div>
</DocumentFragment>
`;

exports[`small left view should support previous price attribute 1`] = `
<DocumentFragment>
  <div
    class="bpk-price"
  >
    <div
      class="bpk-price__leading"
    >
      <span
        class="bpk-text bpk-text--xs bpk-price__previous-price"
      >
        £2,000
      </span>
      <span
        class="bpk-text bpk-text--xs bpk-price__separator"
      >
        𐤟
      </span>
      <span
        class="bpk-text bpk-text--xs"
      >
        from
      </span>
    </div>
    <div>
      <span
        class="bpk-text bpk-text--heading-4 bpk-price__price bpk-price__spacing"
      >
        £1,830
      </span>
      <span
        class="bpk-text bpk-text--xs bpk-price__trailing"
      >
        per day
      </span>
    </div>
  </div>
</DocumentFragment>
`;

exports[`small left view should support trailing text attribute 1`] = `
<DocumentFragment>
  <div
    class="bpk-price"
  >
    <div
      class=""
    />
    <div>
      <span
        class="bpk-text bpk-text--heading-4 bpk-price__price bpk-price__spacing"
      >
        £1,830
      </span>
      <span
        class="bpk-text bpk-text--xs bpk-price__trailing"
      >
        per day
      </span>
    </div>
  </div>
</DocumentFragment>
`;

exports[`small right view should render correctly 1`] = `
<DocumentFragment>
  <div
    class="bpk-price bpk-price--right"
  >
    <div
      class="bpk-price__leading--right"
    />
    <div
      class="bpk-price__column-container"
    >
      <span
        class="bpk-text bpk-text--heading-4 bpk-price__price"
      >
        £1,830
      </span>
    </div>
  </div>
</DocumentFragment>
`;

exports[`small right view should support custom class names 1`] = `
<DocumentFragment>
  <div
    class="bpk-price bpk-price--right custom-classname"
  >
    <div
      class="bpk-price__leading--right"
    />
    <div
      class="bpk-price__column-container"
    >
      <span
        class="bpk-text bpk-text--heading-4 bpk-price__price"
      >
        £1,830
      </span>
    </div>
  </div>
</DocumentFragment>
`;

exports[`small right view should support leading text attribute 1`] = `
<DocumentFragment>
  <div
    class="bpk-price bpk-price--right"
  >
    <div
      class="bpk-price__leading--right"
    >
      <span
        class="bpk-text bpk-text--xs"
      >
        from
      </span>
    </div>
    <div
      class="bpk-price__column-container"
    >
      <span
        class="bpk-text bpk-text--heading-4 bpk-price__price"
      >
        £1,830
      </span>
    </div>
  </div>
</DocumentFragment>
`;

exports[`small right view should support previous price attribute 1`] = `
<DocumentFragment>
  <div
    class="bpk-price bpk-price--right"
  >
    <div
      class="bpk-price__leading bpk-price__leading--right"
    >
      <span
        class="bpk-text bpk-text--xs bpk-price__previous-price"
      >
        £2,000
      </span>
      <span
        class="bpk-text bpk-text--xs bpk-price__separator"
      >
        𐤟
      </span>
      <span
        class="bpk-text bpk-text--xs"
      >
        from
      </span>
    </div>
    <div
      class="bpk-price__column-container"
    >
      <span
        class="bpk-text bpk-text--heading-4 bpk-price__price"
      >
        £1,830
      </span>
      <span
        class="bpk-text bpk-text--xs bpk-price__trailing"
      >
        per day
      </span>
    </div>
  </div>
</DocumentFragment>
`;

exports[`small right view should support trailing text attribute 1`] = `
<DocumentFragment>
  <div
    class="bpk-price bpk-price--right"
  >
    <div
      class="bpk-price__leading--right"
    />
    <div
      class="bpk-price__column-container"
    >
      <span
        class="bpk-text bpk-text--heading-4 bpk-price__price"
      >
        £1,830
      </span>
      <span
        class="bpk-text bpk-text--xs bpk-price__trailing"
      >
        per day
      </span>
    </div>
  </div>
</DocumentFragment>
`;
